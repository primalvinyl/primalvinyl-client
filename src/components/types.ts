export const navList = [
    { path: '/', value: 'Home'},
    { path: '/about-us', value: 'About Us'},
    { path: '/ding', value: 'Nav Item 1'},
    { path: '/dong', value: 'Nav Item 2'},
    { path: '/dingle', value: 'Nav Item 3'},
    { path: '/ring', value: 'Nav Item 4'},
    { path: '/sing', value: 'Nav Item 5'},
];
