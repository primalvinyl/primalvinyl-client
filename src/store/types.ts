/******************************************************************************/
/**************************** Saga Request Types ******************************/
/******************************************************************************/
//request type
export interface RequestType {
    query: string;
    page?: number;
    per_page?: number; 
}

//request default values
export const getRequestDefault: RequestType = {
    query: ''
};

export const getRequestPaginationDefault: RequestType = {
    query: '',
    page: 0,
    per_page: 15
};





/******************************************************************************/
/******************************** Search Query ********************************/
/******************************************************************************/
//action types
export const PUT_QUERY = 'PUT_QUERY';
export const CLEAR_QUERY = 'CLEAR_QUERY';

//action object types
interface PutQueryActionType {
    type: typeof PUT_QUERY;
    payload: string
}

interface ClearQueryActionType {
    type: typeof CLEAR_QUERY;
}

export type QueryActionTypes = PutQueryActionType | ClearQueryActionType;





/******************************************************************************/
/****************************** Song Search ***********************************/
/******************************************************************************/
//action types
export const GET_SONG_SEARCH = 'GET_SONG_SEARCH';
export const PUT_SONG_SEARCH = 'PUT_SONG_SEARCH';
export const CLEAR_SONG_SEARCH = 'CLEAR_SONG_SEARCH';

//action object types
export interface GetSongSearchActionType {
    type: typeof GET_SONG_SEARCH;
    payload: RequestType;
}

interface PutSongSearchActionType {
    type: typeof PUT_SONG_SEARCH;
    payload: SongSearchResultsType;
}

interface ClearSongSearchActionType {
    type: typeof CLEAR_SONG_SEARCH;
}

export type SongSearchActionTypes = PutSongSearchActionType | ClearSongSearchActionType;

//redux store type
interface SongSearchItemType {
    id: number;
    song_title: string;
    song_thumbnail_url: string;
    song_image_url: string;
    artist_name: string;
    artist_image_url: string;
}

export interface SongSearchResultsType {
    results: SongSearchItemType[];
    error: boolean;
    error_message: string;
    request_status: string;
}

//redux store initial value
export const songSearchResultsDefault: SongSearchResultsType = {
    results: [],
    error: false,
    error_message: '',
    request_status: 'idle'
};





/******************************************************************************/
/*********************************** Song *************************************/
/******************************************************************************/
//action types
export const GET_SONG = 'GET_SONG';
export const PUT_SONG = 'PUT_SONG';
export const CLEAR_SONG = 'CLEAR_SONG';

//action object types
export interface GetSongActionType {
    type: typeof GET_SONG;
    payload: RequestType;
}

interface PutSongActionType {
    type: typeof PUT_SONG;
    payload: SongResultType;
}

interface ClearSongActionType {
    type: typeof CLEAR_SONG;
}

export type SongActionTypes = PutSongActionType | ClearSongActionType;

//redux store type
export interface SongResultType {
    id: number;
    song_title: string;
    song_writers: string[];
    song_thumbnail_url: string;
    song_image_url: string;
    artist_name: string;
    artist_image_url: string;
    album_name: string;
    album_image_url: string;
    lyrics: string;
    error: boolean;
    error_message: string;
    request_status: string;
}

//redux store initial value
export const songResultDefault: SongResultType = {
    id: 0,
    song_title: '',
    song_writers: [],
    song_thumbnail_url: '',
    song_image_url: '',
    artist_name: '',
    artist_image_url: '',
    album_name: '',
    album_image_url: '',
    lyrics: '',
    error: false,
    error_message: '',
    request_status: 'idle'
};





/******************************************************************************/
/**************************** Artist Search ***********************************/
/******************************************************************************/
//action types
export const GET_ARTIST_SEARCH = 'GET_ARTIST_SEARCH';
export const PUT_ARTIST_SEARCH = 'PUT_ARTIST_SEARCH';
export const CLEAR_ARTIST_SEARCH = 'CLEAR_ARTIST_SEARCH';

//action object types
export interface GetArtistSearchActionType {
    type: typeof GET_ARTIST_SEARCH;
    payload: RequestType;
}

interface PutArtistSearchActionType {
    type: typeof PUT_ARTIST_SEARCH;
    payload: ArtistSearchResultsType;
}

interface ClearArtistSearchActionType {
    type: typeof CLEAR_ARTIST_SEARCH;
}

export type ArtistSearchActionTypes = PutArtistSearchActionType | ClearArtistSearchActionType;

//redux store type
interface PaginationType {
    page: number;
    pages: number;
    per_page: number;
    items: number;
}

interface ArtistSearchItemType {
    id: number;
    name: string;
    thumbnail_url: string;
    image_url: string;
}

export interface ArtistSearchResultsType {
    pagination: PaginationType,
    results: ArtistSearchItemType[];
    error: boolean;
    error_message: string;
    request_status: string;
}

//redux store initial value
export const paginationDefault: PaginationType = {
    page: 0,
    pages: 1,
    per_page: 15,
    items: 0
};

export const artistSearchResultsDefault: ArtistSearchResultsType = {
    pagination: paginationDefault as PaginationType,
    results: [],
    error: false,
    error_message: '',
    request_status: 'idle'
};





/******************************************************************************/
/********************************* Artist *************************************/
/******************************************************************************/
//action types
export const GET_ARTIST = 'GET_ARTIST';
export const PUT_ARTIST = 'PUT_ARTIST';
export const CLEAR_ARTIST = 'CLEAR_ARTIST';

//action object types
export interface GetArtistActionType {
    type: typeof GET_ARTIST;
    payload: RequestType;
}

interface PutArtistActionType {
    type: typeof PUT_ARTIST;
    payload: ArtistResultType;
}

interface ClearArtistActionType {
    type: typeof CLEAR_ARTIST;
}

export type ArtistActionTypes = PutArtistActionType | ClearArtistActionType;

//redux store type
export interface ArtistResultType {
    name: string;
    realname: string;
    image_url: string;
    error: boolean;
    error_message: string;
    request_status: string;
}

//redux store initial value
export const artistResultDefault: ArtistResultType = {
    name: '',
    realname: '',
    image_url: '',
    error: false,
    error_message: '',
    request_status: 'idle'
};
