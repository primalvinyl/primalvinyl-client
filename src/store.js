import { applyMiddleware, createStore } from 'redux';
import createSagaMiddleware from 'redux-saga';
import rootReducer from './reducers';
import rootSaga from './sagas';

const bindMiddleware = (middleware) => {
    if (process.env.NODE_ENV === 'production') return applyMiddleware(...middleware);
    const { composeWithDevTools } = require('redux-devtools-extension');
    return composeWithDevTools(applyMiddleware(...middleware));
}

const sagaMiddleware = createSagaMiddleware();
const store = createStore(rootReducer, bindMiddleware([sagaMiddleware]));
sagaMiddleware.run(rootSaga);
export default store;
