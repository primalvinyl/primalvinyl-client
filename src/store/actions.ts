import * as types from '../__types__';





/********************************* Action Types *************************************/
export const actionTypes = {
    PUT_QUERY: 'PUT_QUERY',
    PUT_LYRICS_SEARCH: 'PUT_LYRICS_SEARCH',
    PUT_ARTIST_SEARCH: 'PUT_ARTIST_SEARCH',
    PUT_ARTIST: 'PUT_ARTIST',
    CLEAR_LYRICS_SEARCH: 'CLEAR_LYRICS_SEARCH',
    CLEAR_ARTIST_SEARCH: 'CLEAR_ARTIST_SEARCH',
    CLEAR_ARTIST: 'CLEAR_ARTIST',
    GET_LYRICS_SEARCH: 'GET_LYRICS_SEARCH',
    GET_ARTIST: 'GET_ARTIST',
    GET_ARTIST_SEARCH: 'GET_ARTIST_SEARCH'
};





/******************************* Reducer Actions *************************************/
export interface PutQueryActionType {
    type: typeof actionTypes.PUT_QUERY;
    payload: string
}
export const putQuery = (payload: string = ''): PutQueryActionType => {
    return {
        type: actionTypes.PUT_QUERY,
        payload
    }
};




export interface PutLyricsSearchActionType {
    type: typeof actionTypes.PUT_LYRICS_SEARCH;
    payload: types.ReduxLyricsSearchType;
}
export const putLyricsSearch = (payload: types.ReduxLyricsSearchType = types.lyricsSearchDefault): PutLyricsSearchActionType => { 
    return {
        type: actionTypes.PUT_LYRICS_SEARCH,
        payload
    };
};

export interface ClearLyricsSearchActionType {
    type: typeof actionTypes.CLEAR_LYRICS_SEARCH;
}
export const clearGeniusSearch = (): ClearLyricsSearchActionType => { 
    return {
        type: actionTypes.CLEAR_LYRICS_SEARCH
    };
};




export interface PutArtistSearchActionType {
    type: typeof actionTypes.PUT_ARTIST_SEARCH;
    payload: types.ReduxArtistSearchType;
}
export const putArtistSearch = (payload: types.ReduxArtistSearchType = types.artistSearchDefault): PutArtistSearchActionType => { 
    return {
        type: actionTypes.PUT_ARTIST_SEARCH,
        payload
    };
};

export interface ClearArtistSearchActionType {
    type: typeof actionTypes.CLEAR_ARTIST_SEARCH;
}
export const clearArtistSearch = (): ClearArtistSearchActionType => { 
    return {
        type: actionTypes.CLEAR_ARTIST_SEARCH
    };
};




export interface PutArtistActionType {
    type: typeof actionTypes.PUT_ARTIST;
    payload: types.ReduxArtistType;
}
export const putArtist = (payload: types.ReduxArtistType = types.artistDefault): PutArtistActionType => { 
    return {
        type: actionTypes.PUT_ARTIST,
        payload
    };
};

export interface ClearArtistActionType {
    type: typeof actionTypes.CLEAR_ARTIST;
}
export const clearDiscogsArtist = (): ClearArtistActionType => { 
    return {
        type: actionTypes.CLEAR_ARTIST
    };
};





/******************************** Saga Actions **************************************/
export interface GetLyricsSearchActionType {
    type: typeof actionTypes.GET_LYRICS_SEARCH;
    payload: types.SagaRequestType;
}
export const getGeniusSearch = (payload: types.SagaRequestType = types.lyricsGetRequestDefault): GetLyricsSearchActionType => {
    return {
        type: actionTypes.GET_LYRICS_SEARCH,
        payload: { ...types.lyricsGetRequestDefault, ...payload }
    };
};




export interface GetArtistSearchActionType {
    type: typeof actionTypes.GET_ARTIST_SEARCH;
    payload: types.SagaRequestType;
}
export const getDiscogsArtists = (payload: types.SagaRequestType = types.artistGetRequestDefault): GetArtistSearchActionType => {
    return {
        type: actionTypes.GET_ARTIST_SEARCH,
        payload: { ...types.artistGetRequestDefault, ...payload }
    };
}; 




export interface GetDiscogsArtistActionType {
    type: typeof actionTypes.GET_ARTIST;
    payload: types.SagaRequestType;
}
export const getDiscogsArtist = (payload: types.SagaRequestType = types.artistGetRequestDefault): GetDiscogsArtistActionType => {
    return {
        type: actionTypes.GET_ARTIST,
        payload: { ...types.artistGetRequestDefault, ...payload }
    };
};
