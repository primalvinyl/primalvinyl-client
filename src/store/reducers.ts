import { combineReducers } from 'redux';
import * as types from './types';



/******************************** Reducers ***************************************/
export const query = (
    state = '',
    action: any
) => {
    switch (action.type) {
        case types.PUT_QUERY:
            return action.payload
        case types.CLEAR_QUERY:
            return state
        default:
            return state
    }
};


export const songSearchResults = (
    state = types.songSearchResultsDefault,
    action: any
): types.SongSearchResultsType => {
    switch (action.type) {
        case types.PUT_SONG_SEARCH:
            return {
                ...state,
                ...action.payload
            }
        case types.CLEAR_SONG_SEARCH:
            return types.songSearchResultsDefault
        default:
            return state
    }
}


export const song = (
    state = types.songResultDefault,
    action: any
): types.SongResultType => {
    switch (action.type) {
        case types.PUT_SONG:
            return {
                ...state,
                ...action.payload
            }
        case types.CLEAR_SONG:
            return types.songResultDefault
        default:
            return state
    }
}


export const artistSearchResults = (
    state = types.artistSearchResultsDefault,
    action: any
): types.ArtistSearchResultsType => {
    switch (action.type) {
        case types.PUT_ARTIST_SEARCH:
            return {
                ...state,
                ...action.payload
            }
        case types.CLEAR_ARTIST_SEARCH:
            return types.artistSearchResultsDefault
        default:
            return state
    }
}


export const artist = (
    state = types.artistResultDefault,
    action: any
): types.ArtistResultType => {
    switch (action.type) {
        case types.PUT_ARTIST:
            return {
                ...state,
                ...action.payload
            }
        case types.CLEAR_ARTIST:
            return types.artistResultDefault
        default:
            return state
    }
}



/****************************** Root Reducer *************************************/
export default combineReducers({
    query,
    songSearchResults,
    song,
    artistSearchResults,
    artist
});

export interface RootState {
    query: string;
    songSearchResults: types.SongSearchResultsType;
    song: types.SongResultType;
    artistSearchResults: types.ArtistSearchResultsType;
    artist: types.ArtistResultType;
}
