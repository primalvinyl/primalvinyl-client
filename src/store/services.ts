import axios from 'axios';

export const getRequest = (endpoint: string) => {
    return axios.get(endpoint, {
        baseURL: process.env.REACT_APP_API_ENDPOINT,
        // prevents axios from throwing an exception on 404
        validateStatus: () => true
    })
    .then(response => response.data)
    .catch(() => {});
};
