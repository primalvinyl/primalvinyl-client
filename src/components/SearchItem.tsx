import React from 'react';
import { SongResultType } from '../store/types';
import LazyLoadElement from './presentation/LazyLoadElement';
import styles from './SearchItem.module.scss';

const SearchItem = ({ item: {
    song_title,
    artist_name,
    song_writers,
    song_release_date,
    album_name,
    song_image_url,
    lyrics,
    error,
    media_spotify_track_id,
    media_soundcloud_track_id
} }: SearchItemProps): React.ReactElement => {

    const songWriters = song_writers.length > 0 ? song_writers.join(', ') : '';

    // get soundcloud widget endpoint
    let soundcloudEndpoint = '';
    if (media_soundcloud_track_id) {
        const endpoint = process.env.REACT_APP_SOUNDCLOUD_WIDGET_ENDPOINT || '';
        soundcloudEndpoint = endpoint.replace('insertId', media_soundcloud_track_id);
    };

    // get spotify widget endpoint
    let spotifyEndpoint = '';
    if (media_spotify_track_id) {
        const endpoint = process.env.REACT_APP_SPOTIFY_WIDGET_ENDPOINT || '';
        spotifyEndpoint = endpoint.replace('insertId', media_spotify_track_id);
    };

    return (
        <div className={styles.root}>
            {!error &&
                <div>
                    <div className={styles.headerRoot}>
                        <div className={styles.headerWrapper}>
                            <div className={styles.headerImage}>
                                <LazyLoadElement> 
                                    {(onload: any) => 
                                        <img
                                            src={song_image_url}
                                            alt="Song Art"
                                            onLoad={onload}
                                        />
                                    }
                                </LazyLoadElement>
                            </div>
                            <div className={styles.headerMain}>
                                <h2>{song_title}</h2>
                                <div className={styles.headerDetails}>
                                    {artist_name &&
                                        <p>
                                            <span>Artist</span>
                                            <strong>{artist_name}</strong>
                                        </p>
                                    }
                                    {songWriters &&
                                        <p>
                                            <span>Written by</span>
                                            <strong>{songWriters}</strong>
                                        </p>
                                    }
                                    {album_name &&
                                        <p>
                                            <span>Album</span>
                                            <strong>{album_name}</strong>
                                        </p>
                                    }
                                    {song_release_date &&
                                        <p>
                                            <span>Released</span>
                                            <strong>{song_release_date}</strong>
                                        </p>
                                    }
                                </div>
                            </div>
                        </div>
                    </div>
                    <div className={styles.contentRoot}>
                        <div className={styles.contentWrapper}>
                            <div>
                                <h3>Lyrics</h3>
                                <p dangerouslySetInnerHTML={ {__html: lyrics} }></p>
                            </div>
                            {(media_soundcloud_track_id || media_spotify_track_id) &&
                                <div>
                                    <h3>Listen</h3>
                                    {media_soundcloud_track_id &&
                                        <div>
                                            <h4 className={styles.soundcloud}>SoundCloud</h4>
                                            <iframe title="soundcloud" src={soundcloudEndpoint} width="400" height="109" frameBorder="no" scrolling="no" allow="autoplay"></iframe>
                                        </div>
                                    }
                                    {media_spotify_track_id &&
                                        <div>
                                            <h4 className={styles.spotify}>Spotify</h4>
                                            <iframe title="spotify" src={spotifyEndpoint} width="400" height="80" frameBorder="no" allowTransparency={true} allow="encrypted-media"></iframe>
                                        </div>
                                    }
                                </div>
                            }
                        </div>
                    </div>
                </div>
            }
            {error && <h2 className={styles.itemError}>Song data not found</h2>}
        </div>
    );
};

type SearchItemProps = {
    readonly item: SongResultType;
}

export default SearchItem;
