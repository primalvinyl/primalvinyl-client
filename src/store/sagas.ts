import { take, put, call, fork, all } from 'redux-saga/effects';
import { getRequest } from './services';
import * as actions from './actions';
import * as types from './types';
import * as reducers from './reducers';




/******************************** Workers *************************************/
export function* getSongSearchWorker(payload: types.RequestType = types.getRequestDefault): any {
    const { query } = payload;
    try {
        yield put(reducers.query.actions.putQuery(query));
        // start request
        yield put(reducers.songSearch.actions.putSongs({
            ...types.songSearchResultsDefault,
            request_status: 'pending'
        }));
        // make request
        const response = yield call(
            getRequest,
            `/songs/search/${query}`
        );
        // resolve request
        yield put(reducers.songSearch.actions.putSongs({
            ...response,
            request_status: 'resolved'
        }));
    } catch (error) {
        yield put(reducers.songSearch.actions.putSongs({
            ...types.songSearchResultsDefault,
            request_status: 'resolved',
            error: true
        }));
    }
}

export function* getSongWorker(payload: types.RequestType = types.getRequestDefault): any {
    const { query } = payload;
    try {
        // start request
        yield put(reducers.song.actions.putSong({
            ...types.songResultDefault,
            request_status: 'pending'
        }));
        // make request
        const response = yield call(
            getRequest,
            `/songs/${query}`
        );
        // resolve request
        yield put(reducers.song.actions.putSong({
            ...response,
            request_status: 'resolved'
        }));
    } catch (error) {
        yield put(reducers.song.actions.putSong({
            ...types.songResultDefault,
            request_status: 'resolved',
            error: true
        }));
    }
}

export function* getArtistSearchWorker(
    payload: types.RequestType = types.getRequestPaginationDefault
): any {
    const { query, page, per_page } = payload;
    try {
        yield put(reducers.query.actions.putQuery(query));
        // start request
        yield put(reducers.artistSearch.actions.putArtists({
            ...types.artistSearchResultsDefault,
            request_status: 'pending'
        }));
        // make request
        const response = yield call(
            getRequest, 
            `/artists/search/${query}?page=${page}&per_page=${per_page}`
        );
        // resolve request
        yield put(reducers.artistSearch.actions.putArtists({
            ...response,
            request_status: 'resolved'
        }));
    } catch (error) {
        yield put(reducers.artistSearch.actions.putArtists({
            ...types.artistSearchResultsDefault,
            request_status: 'resolved',
            error: true
        }));
    }
}

export function* getArtistWorker(payload: types.RequestType = types.getRequestDefault): any {
    const { query } = payload;
    try {
        yield put(reducers.query.actions.putQuery(query));
        // start request
        yield put(reducers.artist.actions.putArtist({
            ...types.artistResultDefault,
            request_status: 'pending'
        }));
        // make request
        const response = yield call(
            getRequest,
            `/artists/${query}`
        );
        // resolve request
        yield put(reducers.artist.actions.putArtist({
            ...response,
            request_status: 'resolved'
        }));
    } catch (error) {
        yield put(reducers.artist.actions.putArtist({
            ...types.artistResultDefault,
            request_status: 'resolved',
            error: true
        }));
    }
}




/******************************* Watchers *************************************/
export function* getSongSearchWatcher() {
    while (true) {
        const { payload } = yield take(actions.getSongSearch.toString());
        yield call(getSongSearchWorker, payload)
    }
}

export function* getSongWatcher() {
    while (true) {
        const { payload } = yield take(actions.getSong.toString());
        yield call(getSongWorker, payload)
    }
}

export function* getArtistSearchWatcher() {
    while (true) {
        const { payload } = yield take(actions.getArtistSearch.toString());
        yield call(getArtistSearchWorker, payload)
    }
}

export function* getArtistWatcher() {
    while (true) {
        const { payload } = yield take(actions.getArtist.toString());
        yield call(getArtistWorker, payload)
    }
}




/******************************* Root Saga ************************************/
export default function* rootSaga() {
    yield all([
        fork(getSongSearchWatcher),
        fork(getSongWatcher),
        fork(getArtistWatcher),
        fork(getArtistSearchWatcher)
    ]);
} 
