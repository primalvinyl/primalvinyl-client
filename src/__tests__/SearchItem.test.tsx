import React from 'react';
import { shallow } from 'enzyme';
import Renderer from 'react-test-renderer';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import rootReducer from '../store/reducers';
import SearchItem from '../components/SearchItem';
import { songResultDefault } from '../store/types';

describe('SearchItem', () => {
    const store = createStore(rootReducer, {});
    const searchItem = songResultDefault;

    it('renders', () => {
        const myWrapper = shallow(
            <Provider store={store}>
                <MemoryRouter>
                    <SearchItem item={searchItem} />
                </MemoryRouter>
            </Provider>
        );
        expect(myWrapper.exists());
    });
    
    it('matches snapshot', () => {
        const component = Renderer.create(
            <Provider store={store}>
                <MemoryRouter>
                    <SearchItem item={searchItem} />
                </MemoryRouter>
            </Provider>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
