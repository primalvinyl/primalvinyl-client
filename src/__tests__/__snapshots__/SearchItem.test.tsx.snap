// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchItem matches snapshot 1`] = `
<div
  className="root"
>
  <div>
    <div
      className="headerRoot"
    >
      <div
        className="headerWrapper"
      >
        <div
          className="headerImage"
        >
          <div
            className="lazyload-wrapper"
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          >
            <div
              className="lazyload-placeholder"
              style={
                Object {
                  "height": undefined,
                }
              }
            />
          </div>
        </div>
        <h2>
          
        </h2>
        <div
          className="headerDetails"
        >
          <p>
            <span>
              Artist
            </span>
            <strong>
              
            </strong>
          </p>
          <p>
            <span>
              Written by
            </span>
            <strong>
              
            </strong>
          </p>
          <p>
            <span>
              Album
            </span>
            <strong>
              
            </strong>
          </p>
          <p>
            <span>
              Released
            </span>
            <strong>
              
            </strong>
          </p>
        </div>
      </div>
    </div>
    <div
      className="contentRoot"
    >
      <div
        className="contentWrapper"
      >
        <h3>
          Lyrics
        </h3>
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
